sklearn/metrics/tests/test_classification.py

```diff
@@ -1514,30 +1516,86 @@ def test_precision_recall_f1_score_with_an_empty_prediction():
 
 @pytest.mark.parametrize('beta', [1])
 @pytest.mark.parametrize('average', ["macro", "micro", "weighted", "samples"])
-def test_precision_recall_f1_no_labels(beta, average):
+@pytest.mark.parametrize('zero_division', [0, 1])
+def test_precision_recall_f1_no_labels(beta, average, zero_division):
     y_true = np.zeros((20, 3))
     y_pred = np.zeros_like(y_true)
 
-    p, r, f, s = assert_warns(UndefinedMetricWarning,
-                              precision_recall_fscore_support,
-                              y_true, y_pred, average=average,
-                              beta=beta)
+    p, r, f, s = assert_no_warnings(precision_recall_fscore_support, y_true,
+                                    y_pred, average=average, beta=beta,
+                                    zero_division=zero_division)
+    fbeta = assert_no_warnings(fbeta_score, y_true, y_pred, beta=beta,
+                               average=average, zero_division=zero_division)
+
+    zero_division = float(zero_division)
+    assert_almost_equal(p, zero_division)
+    assert_almost_equal(r, zero_division)
+    assert_almost_equal(f, zero_division)
+    assert s is None
+
+    assert_almost_equal(fbeta, float(zero_division))
+
+
+@pytest.mark.parametrize('beta', [1])
+@pytest.mark.parametrize('average', ["macro", "micro", "weighted", "samples"])
+def test_precision_recall_f1_no_labels_warn(beta, average):
+    y_true = np.zeros((20, 3))
+    y_pred = np.zeros_like(y_true)
+
+    func = precision_recall_fscore_support
+    p, r, f, s = assert_warns(UndefinedMetricWarning, func, y_true, y_pred,
+                              average=average, beta=beta)
+    fbeta = assert_warns(UndefinedMetricWarning, fbeta_score, y_true, y_pred,
+                         average=average, beta=beta)
+
     assert_almost_equal(p, 0)
     assert_almost_equal(r, 0)
     assert_almost_equal(f, 0)
     assert s is None
 
-    fbeta = assert_warns(UndefinedMetricWarning, fbeta_score,
-                         y_true, y_pred,
-                         beta=beta, average=average)
     assert_almost_equal(fbeta, 0)
 
 
-def test_precision_recall_f1_no_labels_average_none():
+@pytest.mark.parametrize('zero_division', [0, 1])
```

Comment:
Given your previous comment:

```This is obfuscated. I'd rather a clear, separate test checking the behaviour of zero_divison, than a tiny, unexplicit piece in a larger test.```

I decided to separate this into two tests. I think it is a lot more readable. Otherwise, there are if's or the use of functools.partial. I can go back to previous version, but honestly, if we want readability I think this is better (maybe with better names)